src/redis-cli.c

```diff
@@ -2923,7 +2926,8 @@ static int clusterManagerGetAntiAffinityScore(clusterManagerNodeArray *ipnodes,
             sds name = (sds) dictGetKey(entry);
             int typeslen = sdslen(types);
             if (typeslen < 2) continue;
-            if (types[0] == 'm') score += (10000 * (typeslen - 1));
+            if (types[0] == 'm')
+                for (j = 0; j < typeslen && types[j] == 'm'; ++j) score += 10000;
```

Comment:
@yatsukhnenko redis-cluster doesn't (currently) support one master, it indeed requires at least 3, so your attempt to use with the same address 3 times is invalid. redis-sentinel might be what you need for that use case.

I can see that the loop you added here (searching for `m`) and the `break` you removed below, are a result of this invalid use (a "replica set" that has more than one master), so i think these should be reverted.
maybe instead this case should be detected earlier, and an error message printed to alert the user.

The other changes you did (replacing sdscatprintf with sdscatsds) seems valid, and i wonder why no one noticed it so far. or am i missing some detail that explains why this would never happen on valid use cases?

can you please confirm that this change (the `sdscatsds`) alone (together with the fix in clusterManagerOptimizeAntiAffinity) are enough to avoid the crash?
i.e. that reverting the `for` loop you added and the `break` you removed, solve the crash?

if you can find a proper place to alert the user on the invalid usage please include it in this PR too.
Thanks.

Reply:
@oranagra I'm not going to use cluster with one node, it was my edge (and may be impossible) case when all nodes are down and kubernete's replica set creates cluster from one node and add another nodes. FYI redis-cli creates cluster using the same host three times :)
I added `for` loop there to handle situation when, for example, two masters and three slaves are hosted on the same address, this will produce type like `mmsss` and score will be invalid without my changes.
Replacing `sdscatprintf` with `sdscatsds` also required for correct score calculation and doesn't affect the crash.
In the nutshell only check for `offending_len` is necessary for fixing division by zero bug.

Reply:
@oranagra I can revert all my changes and leave only one line that actually fix crash and you will apply other changes if necessary.

Reply:
ohh, sorry, i got fixated on what you said about creating a cluster with a single node and using same address three times.

so the division by zero was due to the above (using one node 3 times), and the other 3 fixes you just reverted were about a different issue which happens on valid use cases?
in that case, let's fix them, i just wanna be clear on what we fix and why...

(i'm sorry for my short attention span, trying to jiggle between many things)

Reply:
Now this PR contains only fix for division by zero crash.
I'll create another PR with fixes for `clusterManagerGetAntiAffinityScore` and we could discuss that changes separately